lib/common/grpc-service.js

```diff
@@ -446,26 +447,43 @@ GrpcService.createDeadline_ = function(timeout) {
 };
 
 /**
- * Checks for a grpc error code and extends the Error object with additional
+ * Checks for a grpc status code and extends the supplied object with additional
  * information.
  *
  * @private
  *
- * @param {error} err - The original request error.
- * @return {error|null}
+ * @param {object} obj - The object to be extended.
+ * @param {object} response - The grpc response.
+ * @return {object|null}
  */
-GrpcService.getError_ = function(err) {
-  if (err && GRPC_ERROR_CODE_TO_HTTP[err.code]) {
-    var defaultErrorDetails = GRPC_ERROR_CODE_TO_HTTP[err.code];
-    err.code = defaultErrorDetails.code;
-    err.message = err.message || defaultErrorDetails.message;
-    return err;
+GrpcService.extendGrpcResponse_ = function(obj, response) {
```

Comment:
I wasn't necessarily trying to appease you, but I was felt like we were bike shedding. None of the solutions felt very different to me - they all seemed to be fairly intuitive and solved our problem, hence why I compromised.